src/Files.App/ViewModels/Properties/SecurityAdvancedViewModel.cs

```diff
@@ -1,218 +1,251 @@
+		public IAsyncRelayCommand RemoveAccessControlEntryCommand { get; set; }
 
-		public SecurityAdvancedViewModel(ListedItem item, Window window)
+		public SecurityAdvancedViewModel(PropertiesPageNavigationParameter parameter)
 		{
-			IsFolder = item.PrimaryItemAttribute == StorageItemTypes.Folder && !item.IsShortcut;
-			Item = item;
-			Window = window;
-
-			InitializeCommands();
-			GetAccessControlList();
-		}
+			_navigationParameter = parameter;
+			_window = parameter.Window;
 
-		public SecurityAdvancedViewModel(DriveItem item, Window window)
-		{
-			IsFolder = true;
-			Item = new ListedItem()
+			switch (parameter.Parameter)
 			{
-				ItemNameRaw = item.Text,
-				ItemPath = item.Path,
-				PrimaryItemAttribute = StorageItemTypes.Folder
+				case ListedItem listedItem:
+					_path = listedItem.ItemPath;
+					_isFolder = listedItem.PrimaryItemAttribute == StorageItemTypes.Folder && !listedItem.IsShortcut;
+					break;
+				case DriveItem driveItem:
+					_path = driveItem.Path;
+					_isFolder = true;
+					break;
+				default:
+					var defaultlistedItem = (ListedItem)parameter.Parameter;
+					_path = defaultlistedItem.ItemPath;
+					_isFolder = defaultlistedItem.PrimaryItemAttribute == StorageItemTypes.Folder && !defaultlistedItem.IsShortcut;
+					break;
 			};
-			Window = window;
 
-			InitializeCommands();
-			GetAccessControlList();
+			LoadShieldIconResource();
+
+			LoadAccessControlEntry();
+
+			ChangeOwnerCommand = new AsyncRelayCommand(ExecuteChangeOwnerCommand);
+			AddAccessControlEntryCommand = new AsyncRelayCommand(ExecuteAddAccessControlEntryCommand);
+			RemoveAccessControlEntryCommand = new AsyncRelayCommand(ExecuteRemoveAccessControlEntryCommand);
 		}
 
-		private void InitializeCommands()
+		private void LoadShieldIconResource()
 		{
-			ChangeOwnerCommand = new RelayCommand(ChangeOwner, () => AccessControlList is not null);
-			AddAccessControlEntryCommand = new RelayCommand(AddAccessControlEntry, () => AccessControlList is not null && AccessControlList.IsValid);
-			RemoveAccessControlEntryCommand = new RelayCommand(RemoveAccessControlEntry, () => AccessControlList is not null && AccessControlList.IsValid && SelectedAccessControlEntry is not null);
-			DisableInheritanceCommand = new RelayCommand(DisableInheritance, () => AccessControlList is not null && AccessControlList.IsValid && (AccessControlList.IsProtected != _isProtected));
-			SetDisableInheritanceOptionCommand = new RelayCommand<string>(SetDisableInheritanceOption);
-			ReplaceChildPermissionsCommand = new RelayCommand(ReplaceChildPermissions, () => AccessControlList is not null && AccessControlList.IsValid);
+			string imageres = System.IO.Path.Combine(CommonPaths.SystemRootPath, "System32", "imageres.dll");
+			var imageResList = Shell.Win32API.ExtractSelectedIconsFromDLL(
+				imageres,
+				new List<int>() { Constants.ImageRes.ShieldIcon },
+				16);
+
+			ShieldIconFileInfo = imageResList.First();
```

Comment:
I did some work around this in #12068 that might be able to be in common for the shield icon.

Reply:
I have to get it as byte[]. But that public helper method will return other than byte[].

Reply:
Alright, let's keep it as is for now, I'll see if I can do a refactor for that!